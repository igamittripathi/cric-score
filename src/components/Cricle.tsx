import React from 'react';
import { makeStyles, createStyles, Theme, useTheme, styled } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';

interface IResultProps {
    result?: number | string;
}

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            display: 'flex',
            '& > *': {
                margin: theme.spacing(1)
            },
        },
        large: {
            width: theme.spacing(40),
            height: theme.spacing(40),
            fontSize: '125px'
        },
    }),
);

export const Cricle: React.FC<IResultProps> = React.memo<IResultProps>(({ result = 0}) => {
    const classes = useStyles(result);

    return (
        <div className={classes.root}>
            <Avatar className={classes.large}>{result}</Avatar>
        </div>
    );
})

