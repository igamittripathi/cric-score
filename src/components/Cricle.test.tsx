import * as React from 'react';
import { Cricle } from './Cricle';
import { shallow, mount } from "enzyme";
import { Avatar } from '@material-ui/core';


describe("Cricle", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(<Cricle result={0} />);
    })


    test("Cricle should render properly", () => {
        wrapper = shallow(<Cricle/>);
        expect(wrapper).toMatchSnapshot();
    });

    test("Cricle should render Avatar", () => {
        expect(wrapper.find(Avatar)).toHaveLength(1);
    });

    test("allows us to set props", () => {
        expect(wrapper.props().result).toEqual(0);
        wrapper.setProps({ result: 6 });
        expect(wrapper.props().result).toEqual(6);
    });

});