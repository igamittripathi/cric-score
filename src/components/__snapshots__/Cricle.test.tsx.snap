// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BowlingDetail BatingDetail should render properly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-large-2"
  >
    0
  </WithStyles(ForwardRef(Avatar))>
</div>
`;

exports[`Cricle Cricle should render properly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-large-2"
  >
    0
  </WithStyles(ForwardRef(Avatar))>
</div>
`;
